frontend/src/metabase/components/SaveQuestionModal.jsx

```diff
@@ -13,11 +13,14 @@ export default class SaveQuestionModal extends Component {
 
     constructor(props, context) {
         super(props, context);
+
+        const isStructured = Query.isStructured(props.card.dataset_query);
+
         this.state = {
             error: null,
             valid: false,
             details: {
-                name: props.card.name || Query.generateQueryDescription(props.tableMetadata, props.card.dataset_query.query),
+                name: props.card.name || isStructured ? Query.generateQueryDescription(props.tableMetadata, props.card.dataset_query.query) : "",
```

Comment:
it would probably help to put some parens here because otherwise it's not immediately clear whether this means

```javascript
(props.card.name || isStructured) ? Query.generateQueryDescription(props.tableMetadata, props.card.dataset_query.query) : ""
```

or
 
```javascript
props.card.name || (isStructured ? Query.generateQueryDescription(props.tableMetadata, props.card.dataset_query.query) : "")
```